Automattic/woocommerce-payments
# Add seller_message to failed order notes

Fixes #8642  #9156 #9810 

#### Changes proposed in this Pull Request

- Add the new `merchant_message` property to `API_Exception`. This message should not be shown to shoppers.
- When `WC_Payments_API_Client::check_response_for_errors` processes, it will check if the `seller_message` property exists and add this property to `API_Exception` for error_code `card_declined`.
- When catching `API_Exception` during payment process, it will add this message to the order note. 









#### Testing instructions





* Use card `4000 0000 0000 0002` for testing. 
* In the checkout, as a shopper, you will still see the generic decline `Your card was declined.`
* However, as a merchant, in the order note, you can see more details like this: 
![Markup on 2024-12-18 at 15:34:41](https://github.com/user-attachments/assets/89e333a7-46c5-4b75-8c34-b1140984391a)


-------------------


**Post merge**
